### PR TITLE
[tech dept] Use robust server-side filters

### PR DESCRIPTION
**Pull request by polomani**
- got rid of some of the client-side filtering in favour of the new filters/endpoints on the API
- use a newer version of the SDK (includes some class renames)

### CODE REVIEW

File: Cognite.Simulator.Utils/ConfigurationLibraryBase.cs [hunk 1]
```diff
@@ -279,8 +273,9 @@ private async Task ReadConfigurations(CancellationToken token)
                 {
                     Filter = new SimulatorRoutineRevisionFilter()
                     {
-                        // TODO filter by created time, simulatorExternalIds, simulatorIntegrationExternalIds
+                        // TODO filter by created time, simulatorIntegrationExternalIds
```
Comment (sagar-thalwar) [inline Cognite.Simulator.Utils/ConfigurationLibraryBase.cs:276]: question(non-blocking): Why do we need to filter by created time?

**Review by sagar-thalwar — APPROVED** [review]
👍

File: Cognite.Simulator.Utils/ConfigurationLibraryBase.cs (hunk 1 above)
Comment (polomani) [reply Cognite.Simulator.Utils/ConfigurationLibraryBase.cs:276]: we don't really have to, but this is a way to limit the amount of resources needed to be downloaded from the API every time. (i.e. don't have to refetch the revision that was already fetched yesterday)
This is not currently implemented in the new routines logic though...

Comment (polomani) [reply Cognite.Simulator.Utils/ConfigurationLibraryBase.cs:276]: It used to be done with the file-powered APIs

**Pull request merged**